Userland/Libraries/LibAudio/Output.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023, Gregory Bertilson <zaggy1024@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/AtomicRefCounted.h>
+#include <AK/Queue.h>
+#include <AK/Time.h>
+#include <LibAudio/OutputImplementation.h>
+#include <LibAudio/SampleFormats.h>
+#include <LibThreading/ConditionVariable.h>
+#include <LibThreading/MutexProtected.h>
+#include <LibThreading/Thread.h>
+
+namespace Audio {
+
+enum class OutputState {
+    Playing,
+    Suspended,
+};
+
+class Output : public AtomicRefCounted<Output> {
+public:
+    using TaskCompletionCallback = Function<void(Duration)>;
+
+    // Creates a new audio Output class.
+    //
+    // The initial_output_state parameter determines whether it will begin playback immediately.
+    //
+    // The AudioDataRequestCallback will be called when the Output needs more audio data to fill
+    // its buffers and continue playback.
+    //
+    // NOTE: The outputs will retain the previously set volume when running on the PulseAudio
+    //       backend. To start at 100% volume, call set_volume() immediately after creation.
```

Comment:
Maybe set the volume manually in the wrapper class so that differences between backends don't bubble up? New AudioServer connections start at 100% in every case currently, though that might change.